### PR TITLE
New create event

### PR DESCRIPTION
**Pull request by ReggieStar11**
## Why was your change needed?

So the preview now can look polished and done

### CODE REVIEW

File: frontend/components/events/create/CreateEventStepper.tsx [hunk 1]
```diff
@@ -24,67 +24,71 @@ export default function CreateEventStepper({
     <Stepper activeStep={activeStep}>
       <Step onClick={() => {}}>
         <UserIcon className="h-5 w-5" />
-        <div className="absolute -bottom-[4.5rem] w-max text-center">
+        <div className={`absolute -bottom-[4.5rem] w-max text-center ${activeStep === 0 ? '' : 'hidden sm:flex sm:flex-col items-center'}`}>
```
Comment (owya490) [inline frontend/components/events/create/CreateEventStepper.tsx:27]: /nit - can we use double quotes please

File: frontend/components/events/create/CreateEventStepper.tsx [hunk 2]
```diff
@@ -24,67 +24,71 @@ export default function CreateEventStepper({
...
           <Typography
             variant="h6"
             color={activeStep === 0 ? "blue-gray" : "gray"}
+            className={activeStep === 0 ? '' : 'hidden sm:inline'}
```
Comment (owya490) [inline frontend/components/events/create/CreateEventStepper.tsx:31]: if we hide the div above, why do we need to hide it in the child divs as well, line 31 and 37

File: frontend/components/events/create/DescriptionImageForm.tsx [hunk 3]
```diff
@@ -19,12 +20,38 @@ export function DescriptionImageForm({
...
   const updateDescription = (e: string) => {
     updateField({
       description: e,
     });
   };
 
+  const fetchPreviousImagesFromFirebase = async () => {
+    // Fetch previous images from Firebase and return their URLs
+    // Example: You can use Firebase Firestore or Firebase Realtime Database to store image URLs
+    const images = await fetch('YOUR_FIREBASE_IMAGE_URLS_ENDPOINT');
```
Comment (owya490) [inline frontend/components/events/create/DescriptionImageForm.tsx:41]: how does this work? is this meant to be a dud endpoint? we don't fetch from firestore with `http fetches`? we use the firestore sdk in the service layer. 

PS. this function could be moved to `imageService`

File: frontend/components/events/create/DescriptionImageForm.tsx [hunk 4]
```diff
@@ -47,7 +74,7 @@ export function DescriptionImageForm({
             <h4 className="absolute -top-3 left-3 text-sm px-1 bg-white">
               Image
             </h4>
-            {imagePreviewUrl !== "" && (
+            {imagePreviewUrl !== '' && (
```
Comment (owya490) [inline frontend/components/events/create/DescriptionImageForm.tsx:77]: double quotes please!

File: frontend/components/events/create/DescriptionImageForm.tsx [hunk 5]
```diff
@@ -19,12 +20,38 @@ export function DescriptionImageForm({
...
+    const data = await images.json();
+    return data.urls;
+  };
+
+  const handleImageSelection = (imageUrl: string) => {
+    // Set the selected image URL as the preview URL
+    setImagePreviewUrl(imageUrl);
+    // Optionally, you can set the image field in your form data
+    updateField({
+      image: undefined, // Clear the image field since the user selected a previous image
```
Comment (owya490) [inline frontend/components/events/create/DescriptionImageForm.tsx:51]: why do we make this undefined?

File: frontend/components/events/create/DescriptionImageForm.tsx [hunk 6]
```diff
@@ -56,15 +83,32 @@ export function DescriptionImageForm({
...
+            </div>
             <input
-              className="rounded-md ml-4"
+              className="rounded-md mt-4 ml-4"
               accept="image/*"
               type="file"
               onChange={(e) => {
                 if (e.target.files !== null) {
-                  setImagePreviewUrl(URL.createObjectURL(e.target.files![0]));
+                  setImagePreviewUrl(URL.createObjectURL(e.target.files[0]));
```
Comment (owya490) [inline frontend/components/events/create/DescriptionImageForm.tsx:109]: why do you remove this `!`??

File: frontend/components/events/create/DescriptionRichTextEditor.tsx [hunk 7]
```diff
@@ -18,22 +12,16 @@ export default function DescriptionRichTextEditor({
...
-      StarterKit,
-      Underline,
-      Link,
-      Superscript,
-      SubScript,
-      Highlight,
-      TextAlign.configure({ types: ["heading", "paragraph"] }),
-      Placeholder.configure({ placeholder: "Write your description in here!" }),
-    ],
+    extensions: [StarterKit, Link],
```
Comment (owya490) [inline frontend/components/events/create/DescriptionRichTextEditor.tsx:15]: why are we removing all these extensions?

File: frontend/components/events/create/forms/BasicForm.tsx [hunk 8]
```diff
@@ -170,72 +184,56 @@ export function BasicInformation({
           </div>
         </div>
         <div>
-          <label className="text-black text-lg font-semibold">Is your event Private?</label>
```
Comment (owya490) [inline frontend/components/events/create/forms/BasicForm.tsx:173]: why do you delete this?

File: frontend/components/events/create/forms/PreviewForm.tsx [hunk 9]
```diff
@@ -14,67 +14,64 @@ type PreviewFormProps = BasicData & {
...
-          {form.startTime}
+    <div className="md:grid md:grid-cols-3 gap-6 mt-20 items-start mb-8">
+      <div className="col-span-1 mt-6">
+        <div className="text-xl lg:text-xl font-semibold mb-2 border-b-2 border-gray-300 pb-1">Name</div>
+        <p className="text-m">{form.name}</p>
+
+        <div className="text-xl lg:text-xl font-semibold mt-6 mb-2 border-b-2 border-gray-300 pb-1">Location</div>
+        <p className="text-m">{form.location}</p>
+
+        <div className="text-xl lg:text-xl font-semibold mt-6 mb-2 border-b-2 border-gray-300 pb-1">
```
Comment (owya490) [inline frontend/components/events/create/forms/PreviewForm.tsx:33]: we want the date as well! not just the time

File: frontend/components/events/create/forms/PreviewForm.tsx [hunk 10]
```diff
@@ -14,67 +14,64 @@ type PreviewFormProps = BasicData & {
...
-      </FormWrapper>
+      <div className="col-span-1 mt-6">
+        <div className="text-xl lg:text-xl font-semibold mb-2 border-b-2 border-gray-300 pb-1">Price</div>
+        <p className="text-m">{form.price}</p>
+
+        <div className="text-xl lg:text-xl font-semibold mt-6 mb-2 border-b-2 border-gray-300 pb-1">Capacity</div>
+        <p className="text-m">{form.capacity}</p>
+
+        <div className="text-xl lg:text-xl font-semibold mt-6 mb-2 border-b-2 border-gray-300 pb-1">Description</div>
+        <div className="text-m" dangerouslySetInnerHTML={{ __html: form.description }}></div>
```
Comment (owya490) [inline frontend/components/events/create/forms/PreviewForm.tsx:56]: what is `dangerouslySetInnerHTML`

Comment (ReggieStar11) [reply frontend/components/events/create/forms/PreviewForm.tsx:56]: only way to set certian html custom when we preview

File: frontend/components/events/create/CreateEventStepper.tsx (hunk 1 above)
Comment (ReggieStar11) [reply frontend/components/events/create/CreateEventStepper.tsx]: done

File: frontend/components/events/create/CreateEventStepper.tsx (hunk 2 above)
Comment (ReggieStar11) [reply frontend/components/events/create/CreateEventStepper.tsx:31]: there a slight issue if we dont do this line where the stepper loses the first step

File: frontend/components/events/create/DescriptionRichTextEditor.tsx (hunk 7 above)
Comment (ReggieStar11) [reply frontend/components/events/create/DescriptionRichTextEditor.tsx]: added them back, i think when i did it for smaller screens i thought we didnt need them lol

File: frontend/components/events/create/forms/BasicForm.tsx (hunk 8 above)
Comment (ReggieStar11) [reply frontend/components/events/create/forms/BasicForm.tsx:173]: mb added back

File: frontend/components/events/create/forms/PreviewForm.tsx (hunk 9 above)
Comment (ReggieStar11) [reply frontend/components/events/create/forms/PreviewForm.tsx:33]: yessir added in

**Pull request closed without merging**